Failed tests (11684): TestBaseQueueAddRemove

The following test appears to have failed:

[#11684](https://circleci.com/gh/cockroachdb/cockroach/11684):

```
=== RUN   TestBaseQueueAdd
--- PASS: TestBaseQueueAdd (0.01s)
=== RUN   TestBaseQueueProcess
--- PASS: TestBaseQueueProcess (0.01s)
=== RUN   TestBaseQueueAddRemove
panic: test timed out after 5m0s

goroutine 524 [running]:
testing.startAlarm.func1()
	/tmp/workdir/go/src/testing/testing.go:703 +0x16b
created by time.goFunc
	/tmp/workdir/go/src/time/sleep.go:129 +0x6e

goroutine 1 [chan receive, 4 minutes]:
testing.RunTests(0x1cd06f8, 0x23ff200, 0xe0, 0xe0, 0x1)
	/tmp/workdir/go/src/testing/testing.go:562 +0xafa
testing.(*M).Run(0xc820034f10, 0x1)
	/tmp/workdir/go/src/testing/testing.go:494 +0xe5
github.com/cockroachdb/cockroach/util/leaktest.TestMainWithLeakCheck(0xc820034f10)
	/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:37 +0x2f
github.com/cockroachdb/cockroach/storage_test.TestMain(0xc820034f10)
	/go/src/github.com/cockroachdb/cockroach/storage/main_test.go:36 +0x34
main.main()
	github.com/cockroachdb/cockroach/storage/_test/_testmain.go:508 +0x20a

goroutine 17 [syscall, 4 minutes, locked to thread]:
runtime.goexit()
	/tmp/workdir/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 19 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x27471c0)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1029 +0x76
created by github.com/cockroachdb/cockroach/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:610 +0x117

goroutine 539 [chan send, 4 minutes]:
github.com/cockroachdb/cockroach/storage.TestBaseQueueAddRemove(0xc8202d0000)
	/go/src/github.com/cockroachdb/cockroach/storage/queue_test.go:334 +0x7cc
testing.tRunner(0xc8202d0000, 0x23ff548)
	/tmp/workdir/go/src/testing/testing.go:456 +0xdd
created by testing.RunTests
	/tmp/workdir/go/src/testing/testing.go:561 +0xaa4

goroutine 523 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8201bc000, 0xc8202f2020)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70
FAIL	github.com/cockroachdb/cockroach/storage	300.081s
=== RUN   TestBatchBasics
I0120 07:46:07.778168 667 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN   TestBatchGet
I0120 07:46:07.781444 667 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.00s)
=== RUN   TestBatchMerge
I0120 07:46:07.786873 667 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.00s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

--- a/storage/queue_test.go
+++ b/storage/queue_test.go
@@ -326,12 +326,11 @@ func TestBaseQueueAddRemove(t *testing.T) {
 	bq.MaybeRemove(r)
 
 	// Wake the queue
-	testQueue.blocker <- struct{}{}
+	close(testQueue.blocker)
 
 	// Make sure the queue has actually run through a few times
 	for i := 0; i < cap(bq.incoming)+1; i++ {
 		bq.incoming <- struct{}{}
-		testQueue.blocker <- struct{}{}
 	}
 
 	if pc := atomic.LoadInt32(&testQueue.processed); pc > 0 {

